### PR TITLE
add new topics in week1 readme

### PR DESCRIPTION
**Pull request by ramsha07**

### CODE REVIEW

File: week1/readme.md [hunk 1]
```diff
@@ -1,9 +1,87 @@
...
+
+2. Code writing: 
+Code can be evaluated upon three axes. First, correctness refers to does the code run as intended Second, design refers to how well is the code designed? Finally, style refers to how aesthetically pleasing and consistent is the code?
+
+3. Functions:
+The statement at the start of the code #include <stdio.h> is a very special command that tells the compile that you want to use the capabilities of library called stdio.h. This allows you, among many other things, to utilize the printf function
+Within the printf function it isvery important to note that without certain smybols like the the round brackets after the printf function containg the double quotes at the start and end of the and within thore double quotes we write what we want to print eg; Hello, world followed by a \n to get the next line and ending with the semicolon
```
Comment (syedqasimahmed) [inline week1/readme.md]:
```suggestion
Within the printf function it is very important to note that without certain symbols like the the round brackets after the printf function contains the double quotes at the start and end of the and within there double quotes we write what we want to print eg; Hello, world followed by a \n to get the next line and ending with the semicolon
```

File: week1/readme.md [hunk 2]
```diff
@@ -1,9 +1,87 @@
...
+Execute make hello and then Typing ./hello in the terminal window to run the program.
+
+4. Variables
+#include <cs50.h> should be added to the top of your code, this deals with taking thw input from the user. For taking a string from the user, get_string function is used. Then, the variable answer is passed to the printf function. %s tells the printf function to prepare itself to receive a string follewed by the name of that variable.
```
Comment (syedqasimahmed) [inline week1/readme.md]: whenever you are referring to any code chunk /variable / function name please use this `#include <cs50.h>`

File: week1/readme.md [hunk 3]
```diff
@@ -1,9 +1,87 @@
...
+     printf("x is greater than y\n");
+   }
+    else
+   {
+     printf("x is equal to y\n");
+   }
+Despite of using greater than or lesser than sign we can also use  == ensure that something is equal to something else, where a single equal sign would have a very different function in C. Also the ||  means or, && means and, != means not equal to.
+
+7. Loops:
+```javascript
```
Comment (syedqasimahmed) [inline week1/readme.md]: Only code goes into this tag not every other text 
also it should be changed to 
```c 

```

File: week1/readme.md [hunk 4]
```diff
@@ -1,9 +1,87 @@
...
+4. Dry run:
+A dry run (or practice run) is a software testing process used to make sure that a system works correctly and will not result in severe failure.At the most basic level, a dry run test involves a programmer reading the code to make sure it works without actually running.I learned the dry run by executing my loop code within the dry run  to make sure that the loop is working fine.
+
+5. Truth table:
+Mathematics normally uses a two-valued logic: every statement is either true or false, by using operations like: Conjunction (AND), disjunction (OR), negation (NOT), 
+
+6. Conditionals
+    if/ else/ else if
+    if x is greater than y, if true? the condition if will run or if false i will do something else if that condition is not met.
+    ```javascript
```
Comment (syedqasimahmed) [inline week1/readme.md]: it is not javascript it is c so it should be something like this 
```c
```

File: week1/readme.md (hunk 2 above)
Comment (syedqasimahmed) [inline week1/readme.md]: whenever you are referring to any code chunk /variable / function name please use this `#include <cs50.h>`

**Comment by syedqasimahmed** [comment]
@ramsha07 leaving some comment for reference I am resolving them for now.

File: week1/readme.md (hunk 2 above)
Comment (syedqasimahmed) [reply week1/readme.md]: `printf`

**Review by syedqasimahmed — APPROVED** [review]

File: week1/readme.md (hunk 2 above)
Comment (ramsha07) [reply week1/readme.md]: Acknowledged

Comment (ramsha07) [reply week1/readme.md]: Will look into this, the next time

File: week1/readme.md (hunk 4 above)
Comment (ramsha07) [reply week1/readme.md]: I will make sure the next time

File: week1/readme.md (hunk 2 above)
Comment (ramsha07) [reply week1/readme.md]: in this particular you are talking about the get_string?

**Pull request merged**